Обработка исключений позволяет обрабатывать ошибки и восстанавливаться после них. Для этого используются блоки **try-catch** для инкапсуляции кода, который может выбросить исключение. Если исключение происходит в блоке **try**, то выполняются блок(и) **catch**, подходящие типу исключения, что позволяет правильно его обработать. Код в блоке **finally** выполняется вне зависимости от того, возникло исключение или нет.